test/disk_usage_test.rb

```diff
@@ -0,0 +1,45 @@
+require 'test_helper'
+require 'rack/test'
+
+require 'smart_proxy_pulp_plugin/pulp_plugin'
+require 'smart_proxy_pulp_plugin/disk_usage'
+
+class DiskUsageTest < Test::Unit::TestCase
+  include ::Proxy::Util
+  def test_has_path_should_be_true
+    disk_test = ::PulpProxy::DiskUsage.new(:path => {:root => ::Sinatra::Application.settings.root})
+    assert_equal(::Sinatra::Application.settings.root, disk_test.path)
+  end
+
+  def test_path_can_be_string
+    disk_test = ::PulpProxy::DiskUsage.new(:path => ::Sinatra::Application.settings.root)
+    assert_equal(::Sinatra::Application.settings.root, disk_test.path)
+  end
+
+  def test_raise_error_if_path_nil
+    assert_raises Proxy::Error::ConfigurationError do
+      ::PulpProxy::DiskUsage.new
+    end
+  end
+
+  def test_command_df
+    disk_test = ::PulpProxy::DiskUsage.new(:path => ::Sinatra::Application.settings.root)
+    assert(disk_test.send(:command).include?(which('df')))
+  end
+
+  def test_has_stat_should_be_true
+    disk_test = ::PulpProxy::DiskUsage.new(:path => {:root => ::Sinatra::Application.settings.root})
+    assert_not_nil(disk_test.stat)
+    assert(disk_test.stat.is_a?(Hash))
+  end
+
+  def test_should_return_valid_json
+    disk_test = ::PulpProxy::DiskUsage.new(:path => {:root => ::Sinatra::Application.settings.root})
+    data = disk_test.stat
+    assert_equal([:filesystem, :"1g-blocks", :used, :available, :percent, :mounted, :path, :size], data[:root].keys)
+    json = disk_test.to_json
+    assert_nothing_raised JSON::ParserError do
+      JSON.parse(json)
+    end
+  end
+end
```

Comment:
I miss a test with multiple paths, can you add one? You could use a temp directory as a second one perhaps, we want to test the looping.